src/ui/LoadingScreen.tsx

```diff
@@ -31,27 +31,24 @@ export function LoadingScreen(): React.ReactElement {
   });
 
   useEffect(() => {
-    async function doLoad(): Promise<void> {
-      await load()
-        .then((saveString) => {
-          try {
-            Engine.load(saveString);
-          } catch (err: unknown) {
-            ActivateRecoveryMode();
-            setLoaded(true);
-            throw err;
-          }
-
-          pushGameReady();
-          setLoaded(true);
-        })
-        .catch((reason) => {
-          console.error(reason);
-          Engine.load("");
+    load()
+      .then(async (saveData) => {
```

Comment:
This is the weirdest mishmash of `async` and promise chains. IMO, let's chop out a layer of this.

If we (reasonably) assume that `pushGameReady` and `setLoaded` won't throw (they're quite simple and I've never seen any issues with them), then the final catch is really just catching the result of `throw err`. So it can all be merged into the upper catch as
```js
console.errror(err);
ActivateRecoveryMode();
await Engine.load("");
setLoaded(true);
```